File: frontend/src/api/summary.ts
/** 讨论纪要接口（r008）。
 *
 * 设计事实源：`docs/rounds/r008-assignment-gaps/design.md` §3；后端 `api/routers/summary.py`。
 */
import { request } from './http'

export interface SessionSummary {
  id: string
  roomId: string
  status: 'ready' | 'failed'
  provider: string
  model: string
  inputDigest: string
  content: string
  error: string | null
  createdBy: string | null
  createdAt: string
  updatedAt: string
}

export const summaryApi = {
  /** 查看纪要；未生成时 `summary` 为 null。 */
  get: (roomId: string) => request<{ summary: SessionSummary | null }>(`/api/rooms/${roomId}/summary`),
  /** 生成 / 重新生成（Host/Moderator）。 */
  generate: (roomId: string) => request<SessionSummary>(`/api/rooms/${roomId}/summary`, { method: 'POST' }),
}

File: backend/app/schemas/summary.py
"""纪要相关的 VO（r008）。

设计事实源：`docs/rounds/r008-assignment-gaps/design.md` §2.1；一条房间一份、覆盖式重生（ADR-0018）。
"""
from __future__ import annotations

from datetime import datetime
from typing import Optional

from app.schemas.common import CamelModel


class SessionSummaryVO(CamelModel):
    id: str
    room_id: str
    status: str
    provider: str
    model: str
    input_digest: str
    content: str
    error: Optional[str] = None
    created_by: Optional[str] = None
    created_at: datetime
    updated_at: datetime
